client/driver/lxc.go

```diff
@@ -0,0 +1,486 @@
+	GPGKeyID             string   "mapstructure:`gpg_key_id`"
+	GPGKeyServer         string   "mapstructure:`gpg_key_server`"
+	DisableGPGValidation bool     "mapstructure:`disable_gpg`"
+	FlushCache           bool     "mapstructure:`flush_cache`"
+	ForceCache           bool     "mapstructure:`force_cache`"
+	TemplateArgs         []string "mapstructure:`template_args`"
+	LogLevel             string   `mapstructure:"log_level"`
+	Verbosity            string
+}
+
+// NewLxcDriver returns a new instance of the LXC driver
+func NewLxcDriver(ctx *DriverContext) Driver {
+	return &LxcDriver{DriverContext: *ctx}
+}
+
+// Validate validates the lxc driver configuration
+func (d *LxcDriver) Validate(config map[string]interface{}) error {
+	fd := &fields.FieldData{
+		Raw: config,
+		Schema: map[string]*fields.FieldSchema{
+			"template": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: true,
+			},
+			"distro": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"release": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"arch": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"image_variant": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"image_server": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"gpg_key_id": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"gpg_key_server": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"disable_gpg": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"flush_cache": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"force_cache": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"template_args": &fields.FieldSchema{
+				Type:     fields.TypeArray,
+				Required: false,
+			},
+			"log_level": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+			"verbosity": &fields.FieldSchema{
+				Type:     fields.TypeString,
+				Required: false,
+			},
+		},
+	}
+
+	if err := fd.Validate(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Fingerprint fingerprints the lxc driver configuration
+func (d *LxcDriver) Fingerprint(cfg *config.Config, node *structs.Node) (bool, error) {
+	version := lxc.Version()
+	if version == "" {
+		return false, nil
+	}
+	node.Attributes["driver.lxc.version"] = version
+	node.Attributes["driver.lxc"] = "1"
+	return true, nil
+}
+
+// Start starts the LXC Driver
+func (d *LxcDriver) Start(ctx *ExecContext, task *structs.Task) (DriverHandle, error) {
+	var driverConfig LxcDriverConfig
+	if err := mapstructure.WeakDecode(task.Config, &driverConfig); err != nil {
+		return nil, err
+	}
+	lxcPath := lxc.DefaultConfigPath()
```

Comment:
I am not very sure if we can have multiple Config Paths on a machine, usually all the containers live in `/var/lib/lxc`. Having it in the task dir is desirable since then we can make use of disk space accounting.